tests/Api/Locale/ListLocaleApiIntegration.php

```diff
@@ -0,0 +1,231 @@
+        $this->assertTrue($secondPage->hasPreviousPage());
+        $this->assertTrue($secondPage->hasNextPage());
+        $this->assertSame($baseUri . '/api/rest/v1/locales?page=1&limit=3&with_count=false', $secondPage->getPreviousLink());
+
+        $locales = $secondPage->getItems();
+        $this->assertCount(3 ,$locales);
+        $this->assertSameContent($expectedLocales[3], $locales[0]);
+        $this->assertSameContent($expectedLocales[4], $locales[1]);
+        $this->assertSameContent($expectedLocales[5], $locales[2]);
+
+        $lastPage = $secondPage->getNextPage();
+        $this->assertInstanceOf(PageInterface::class, $lastPage);
+        $this->assertTrue($lastPage->hasPreviousPage());
+        $this->assertFalse($lastPage->hasNextPage());
+        $this->assertNull($lastPage->getNextPage());
+        $this->assertCount(0 ,$lastPage->getItems());
+
+        $previousPage = $lastPage->getPreviousPage();
+        $this->assertInstanceOf(PageInterface::class, $previousPage);
+
+        $locales = $previousPage->getItems();
+        $this->assertCount(3 ,$locales);
+        $this->assertSameContent($expectedLocales[3], $locales[0]);
+        $this->assertSameContent($expectedLocales[4], $locales[1]);
+        $this->assertSameContent($expectedLocales[5], $locales[2]);
+    }
+
+    public function testListPerPageWithCount()
+    {
+        $api = $this->createClient()->getLocaleApi();
+        $baseUri = $this->getConfiguration()['api']['baseUri'];
+
+        $firstPage = $api->listPerPage(3, true);
+        $this->assertInstanceOf(PageInterface::class, $firstPage);
+        $this->assertSame(6, $firstPage->getCount());
+        $this->assertSame($baseUri . '/api/rest/v1/locales?page=2&limit=3&with_count=true', $firstPage->getNextLink());
+    }
+
+    public function testListPerPageWithSpecificQueryParameter()
+    {
+        $api = $this->createClient()->getLocaleApi();
+        $expectedLocales = $this->getExpectedLocales();
+        $baseUri = $this->getConfiguration()['api']['baseUri'];
+
+        $firstPage = $api->listPerPage(2, false, ['foo' => 'bar']);
+
+        $this->assertInstanceOf(PageInterface::class, $firstPage);
+        $this->assertSame($baseUri . '/api/rest/v1/locales?page=2&limit=2&with_count=false&foo=bar', $firstPage->getNextLink());
+
+        $locales = $firstPage->getItems();
+        $this->assertCount(2 ,$locales);
+        $this->assertSameContent($expectedLocales[0], $locales[0]);
+        $this->assertSameContent($expectedLocales[1], $locales[1]);
+    }
+
+    public function testAll()
+    {
+        $api = $this->createClient()->getLocaleApi();
+        $locales = $api->all();
+        $expectedLocales = $this->getExpectedLocales();
+
+        $this->assertInstanceOf(ResourceCursorInterface::class, $locales);
+
+        $localesCount = 0;
+        foreach ($locales as $key => $locale) {
+            $this->assertSameContent($expectedLocales[$key], $locale);
```

Comment:
It check the order because `$key` is numeric, but in fact, the `foreach` itself is useless...